File: Questão9.java
class Autor {
    private String nome;

    public Autor(String nome) {
        this.nome = nome;
    }

    public String getNome() {
        return nome;
    }
}

class Livro {
    private String titulo;
    private Autor autor;

    public Livro(String titulo, Autor autor) {
        this.titulo = titulo;
        this.autor = autor;
    }

    public String detalhes() {
        return titulo + " escrito por " + autor.getNome();
    }

    public static void main(String[] args) {
        Autor autor = new Autor("George Orwell");
        Livro livro = new Livro("1984", autor);
        System.out.println(livro.detalhes()); 
    }
}

File: Questão9.go
package main

import "fmt"


type Autor struct {
    Nome string
}


type Livro struct {
    Titulo string
    Autor  Autor
}


func (l Livro) Detalhes() string {
    return fmt.Sprintf("%s escrito por %s", l.Titulo, l.Autor.Nome)
}

func main() {
    autor := Autor{Nome: "George Orwell"}
    livro := Livro{Titulo: "1984", Autor: autor}
    fmt.Println(livro.Detalhes()) 
}
